### PR TITLE
More structure for the README

### PR DESCRIPTION
**Pull request by zamith**
The README was starting to become hard to read, in my opinion, as it had a few
separate paragraphs that didn't really relate. This tries to group them into
well defined categories.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,19 +7,24 @@ following command:
...
-blog post](http://crystal-lang.org/2016/05/26/heroku-buildpack.html).
+In order for the buildpack to work properly you should have a `shard.yml` file,
+as it is how it will detect that your app is a Crystal app.
 
-Your application has to listen on a port defined by Heroku.
-It is given to you through the command line option `--port` and the environment variable `PORT` (accessible through `ENV["PORT"]` in Crystal).
+Your application has to listen on a port defined by Heroku. It is given to you
+through the command line option `--port` and the environment variable `PORT`
+(accessible through `ENV["PORT"]` in Crystal). Usually you should not have to
+worry about this.
```
Comment (jhass) [inline README.md]: > Usually you should not have to worry about this.

Why not?

Comment (zamith) [inline README.md]: I was thinking about when you are using a framework such as `Kemal` or `Frank`, in which case you don't really have to do anything, I believe. What would you need to worry about?

Comment (jhass) [inline README.md]: Let's be explicit though, "However most web frameworks should handle this for you."

Comment (zamith) [inline README.md]: Yup, that makes sense. I like that wording better.

**Pull request merged**